### PR TITLE
Fix looking up implicit team with conflicts CORE-9732

### PR DESCRIPTION
**Pull request by zapu**
also more debugging capabilities in `keybase riit`

### CODE REVIEW

**Comment by zapu** [comment]
Max I merged your changes, added some logs/tweaks. Notably, it will only retry loading only when there are conflicts. We wouldn't want to retry on complete failures, because we do not cache bad loads. So if someone wants to lookup 100% nonexisting impteam (not even a non-existing conflict of existing impteam), there's no reason to try twice.

**Comment by zapu** [comment]
That being said, maybe there's another way to do it, because it became a bit messy. I'll spend another hour exploring but probably not today.

File: go/teams/implicit.go [hunk 1]
```diff
@@ -113,35 +113,27 @@ func loadImpteamFromServer(ctx context.Context, g *libkb.GlobalContext, displayN
...
 }
 
-// Lookup an implicit team by name like "alice,bob+bob@twitter (conflicted copy 2017-03-04 #1)"
-// Does not resolve social assertions.
-// preResolveDisplayName is used for logging and errors
-func lookupImplicitTeamAndConflicts(ctx context.Context, g *libkb.GlobalContext,
-	preResolveDisplayName string, impTeamNameInput keybase1.ImplicitTeamDisplayName, opts ImplicitTeamOptions) (
-	team *Team, teamName keybase1.TeamName, impTeamName keybase1.ImplicitTeamDisplayName, conflicts []keybase1.ImplicitTeamConflictInfo, err error) {
+// attemptLoadImpteamAndConflits attempts to lead the implicit team with conflict, but it might hit the cache based on
+// the force flag being true. In that case, the caller ought to recall this function with the force flag set to true.
```
Comment (maxtaco) [inline go/teams/implicit.go]: should be `the force flag being false`

**Comment by maxtaco** [comment]
Thanks @zapu. looks good!

**Comment by maxtaco** [comment]
maybe there's an easier fix, let me try it

**Comment by maxtaco** [comment]
See the above commit ^

**Comment by zapu** [comment]
looks perfect. we might even remove ` impTeamName.ConflictInfo.IsConflict()`
from the if condition for retrying now

sob., 22 gru 2018 o 19:52 Maxwell Krohn <notifications@github.com>
napisał(a):

**Comment by maxtaco** [comment]
Good to merge when you are: 👍

File: go/teams/implicit.go [hunk 2]
```diff
@@ -60,34 +60,34 @@ func LookupImplicitTeamAndConflicts(ctx context.Context, g *libkb.GlobalContext,
...
-	imp, err := loadImpteam(ctx, g, displayName, public)
+	imp, _, err := loadImpteam(ctx, g, displayName, public, false /* force */)
 	if err != nil {
 		return res, err
 	}
 	return imp.TeamID, nil
 }
 
-func loadImpteam(ctx context.Context, g *libkb.GlobalContext, displayName string, public bool) (imp implicitTeam, err error) {
+func loadImpteam(ctx context.Context, g *libkb.GlobalContext, displayName string, public bool, force bool) (imp implicitTeam, hitCache bool, err error) {
```
Comment (joshblum) [inline go/teams/implicit.go]: maybe `forceCache` or `useCache` instead? i feel like `force` would typically imply forcing a server call not a cache call so a more explicit name might be less confusing in the future

**Pull request merged**